mdgo/coordination.py

```diff
@@ -419,30 +421,61 @@ def heat_map(
         The coordinates of the floating ion around clusters normalized to the desired cartesian coordinate system.
     """
     dimension = len(dim)
+    if isinstance(cluster_terminal, list):
+        mode = "ordered"
+        if len(cluster_terminal) != dimension:
+            raise ValueError(
+                "Please specify the cluster_terminal in the order of {}.".format(", ".join(c for c in dim))
+            )
+    else:
+        mode = "unordered"
     trj_analysis = nvt_run.trajectory[run_start:run_end:]
     coordinates = []
     for i, ts in enumerate(trj_analysis):
         if cluster_center_sites[i] == 0:
             pass
         else:
             center_atom = nvt_run.select_atoms("index " + str(cluster_center_sites[i]))[0]
-            selection = "(" + cluster_terminal + ") and (same resid as index " + str(center_atom.index) + ")"
-            bind_atoms = nvt_run.select_atoms(selection, periodic=True)
-            if len(bind_atoms) == dimension:
-                vertex_atoms = bind_atoms
-            elif len(bind_atoms) > dimension:
-                distances = distance_array(ts[floating_atom.index], bind_atoms.positions, ts.dimensions)
-                idx = np.argpartition(distances[0], 3)
-                vertex_atoms = bind_atoms[idx[:3]]
+            if mode == "ordered":
+                selections = [
+                    "(" + species + ") and (same resid as index " + str(center_atom.index) + ")"
+                    for species in cluster_terminal
+                ]
+                bind_atoms_xyz = [nvt_run.select_atoms(sel, periodic=True) for sel in selections]
+                vertex_atoms: List[Atom] = list()
+                for atoms in bind_atoms_xyz:
+                    if len(atoms) == 1:
+                        vertex_atoms[i] = atoms[0]
+                    elif len(atoms) > 1:
+                        distances = distance_array(ts[floating_atom.index], atoms.positions, ts.dimensions)
+                        idx = np.argpartition(distances[0], 1)
+                        vertex_atoms[i] = atoms[idx[0]]
+                    else:
+                        raise ValueError(
+                            "There should be at least 1 cluster_terminal atom in the {} dimension."
+                            "Try broadening the selection at index {} of the cluster_terminal ".format(
+                                str(dim[i]), str(i + 1)
+                            )
+                        )
```

Comment:
There's a problem with `vertex_atoms` in here somewhere, I believe caused by the line `vertex_atoms[i] = atoms[0]` because the index i does not exist. I think you either need to use `append` or instantiate the list at the top of the for loop and ensure that something gets added to it every iteration.

As it stands, I get the following failure:

```IndexError: list assignment index out of range```